docs/open-source/events-manager.mdx

```diff
@@ -5,87 +5,60 @@ description: "How events are emitted and consumed."
-
-### Initialization
+## Current Event Types
 
-```python
-def __init__(self, subscriptions: List[EventType] = []):
-    self.queue = asyncio.Queue()
-    self.subscriptions = set(subscriptions)
-    self.active = False
-```
+The current event types include:
 
-The `EventsManager` constructor accepts an optional list of `EventType` subscriptions. By default, it initializes an empty set of subscriptions, an asynchronous queue, and sets the `active` attribute to `False`.
+1. `TRANSCRIPT`: Indicates a partial transcript for the conversation has been received.
+2. `TRANSCRIPT_COMPLETE`: Indicates the transcript is complete (ie conversation has ended).
+3. `ACTION`: Indicates that a Vocode action has begun or completed.
+4. `PHONE_CALL_CONNECTED`: Indicates a phone call has been connected (only gets sent during `PhoneConversation`s)
+5. `PHONE_CALL_ENDED`: Indicates a phone call has ended.
 
-### Publishing Events
+## Usage
 
-```python
-def publish_event(self, event: Event):
-    if event.type in self.subscriptions:
-        self.queue.put_nowait(event)
-```
+Using the events manager to take particular action when events fire requires that you subclass `vocode.streaming.utils.EventsManager` and override the `handle_event` method.
 
-The `publish_event` method takes an `Event` object as input and adds it to the queue if its type is in the set of subscribed event types.
+You can also configure which events your `EventsManager` is subscribed to by using the `subscriptions` property (see example).
 
-### Starting the Event Loop
+### Example
 
 ```python
-async def start(self):
-    self.active = True
-    while self.active:
-        try:
-            event: Event = await self.queue.get()
-        except asyncio.QueueEmpty:
-            await asyncio.sleep(1)
-        self.handle_event(event)
-```
+from vocode.streaming.models.events import Event, EventType
 
-## Current Event Types
+from vocode.streaming.models.events import Event, EventType
+from vocode.streaming.models.transcript import TranscriptCompleteEvent
+from vocode.streaming.utils.events_manager import EventsManager
 
-The current event types include:
 
-1. `TRANSCRIPT`: Indicates a partial transcript for the conversation has been received.
-2. `TRANSCRIPT_COMPLETE`: Indicates the transcript is complete (ie conversation has ended).
-3. `PHONE_CALL_CONNECTED`: Indicates a phone call has been connected.
-4. `PHONE_CALL_ENDED`: Indicates a phone call has ended.
-5. `RECORDING`: (Vonage Only) Indicates a secure URL containing a recording of the call is available. Requires `recording=true` in `VonageConfig`.
+class CustomEventsManager(EventsManager):
+    def __init__(self):
+        super().__init__([EventType.TRANSCRIPT_COMPLETE])
 
-## Example Usage
+    async def handle_event(self, event: Event):
+        if isinstance(event, TranscriptCompleteEvent):
```

Comment:
why do we switch to isinstance here? The `event.type` check is better practice imo

Reply:
If it helps for typechecking for coding then fine to leave as-is tho

Reply:
it's for typechecking